addons/l10n_ch/models/ir_actions_report.py

```diff
@@ -27,11 +27,12 @@ def apply_qr_code_ch_cross_mask(self, width, height, barcode_drawing):
         qr_cross = ReportLabImage((width/2 - cross_width/2) / mm, (height/2 - cross_height/2) / mm, cross_width / mm, cross_height / mm, cross_path.as_posix())
         barcode_drawing.add(qr_cross)
 
-    def _render_qweb_pdf_prepare_streams(self, data, res_ids=None):
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # OVERRIDE
-        res = super()._render_qweb_pdf_prepare_streams(data, res_ids)
-        if res_ids and self.report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
-            invoices = self.env[self.model].browse(res_ids)
+        res = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids)
+        if res_ids and report_ref in ('account.report_invoice_with_payments', 'account.report_invoice'):
+            report = self._get_report(report_ref)
```

Comment:
That seems a bit weird, shouldn't it be e.g. `self._get_report(report_ref).report_name` to correctly handle the various possibilities? Or does this method only allow the "report_name" kind of report, in which case using `_get_report` seems misleading?